util/internal/common/src/main/java/org/hibernate/search/v6poc/util/impl/common/AbstractCloser.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.util.impl.common;
+
+/**
+ * A base class implementing the logic behind {@link Closer} and {@link SuppressingCloser}.
+ *
+ * @param <E> The supertype of exceptions this object can catch
+ *
+ * @author Yoann Rodiere
+ */
+public abstract class AbstractCloser<S, E extends Exception> {
+
+	/**
+	 * Execute the given close {@code operator} <strong>immediately</strong> on
+	 * the given {@code objectToClose}, swallowing any throwable in order to
+	 * {@link Throwable#addSuppressed(Throwable) add it as suppressed} to a previously caught throwable,
+	 * or to re-throw it later.
+	 * @param operator An operator to close {@code objectToClose}. Accepts lambdas
+	 * such as {@code MyType::close}.
+	 * @param objectToClose An object to close
```

Comment:
Same here and for all the iterations of the parameter in the class (and its variation `objectsToClose`).